### PR TITLE
MAID-3015: add integration tests for peer remove

### PR DESCRIPTION
**Pull request by fizyk20**

### CODE REVIEW

File: tests/integration_tests.rs [hunk 1]
```diff
@@ -232,6 +232,40 @@ mod test {
...
+            &mut env,
+            &ScheduleOptions {
+                genesis_size: 6,
+                peers_to_remove: 1,
+                ..Default::default()
+            },
+        );
+
+        let result = env.network.execute_schedule(schedule);
+        assert!(result.is_ok(), "{:?}", result);
```
Comment (pierrechevalier83) [inline tests/integration_tests.rs]: I would like to see a bit more:
* proof that consensus on removing that peer was reached
* proof that if some decisions are taken next, this peer is never a signatory of the consensused block

File: tests/integration_tests.rs [hunk 2]
```diff
@@ -232,6 +232,40 @@ mod test {
...
+                peers_to_remove: 3,
+                // high probability - peers will be removed in close succession
+                prob_remove: 0.2,
+                ..Default::default()
+            },
+        );
+
+        let result = env.network.execute_schedule(schedule);
+        assert!(result.is_ok(), "{:?}", result);
+    }
```
Comment (pierrechevalier83) [inline tests/integration_tests.rs]: Same as above: I would like a bit more granular proof that the Network behaved as expected. Will probably be rather easy once you've made it possible to input the observations to vote for in a certain order.

File: tests/integration_tests.rs (hunk 1 above)
Comment (madadam) [reply tests/integration_tests.rs]: I'd say the first point is probably already taken care of internally by `execute_schedule`. Not sure though. @fizyk20, can you confirm this?

Comment (fizyk20) [reply tests/integration_tests.rs]: Wow, didn't notice your comment. Yes, I'd say so - the check that all observations are consensused should be taking care of it.

Comment (madadam) [reply tests/integration_tests.rs]: I wonder whether this test is even useful anymore. Doesn't the `remove_many_peers_at_once` test cover everything this one does and more?

File: tests/integration_tests.rs [hunk 3]
```diff
@@ -247,6 +250,70 @@ mod test {
...
+        let schedule = Schedule::new(
+            &mut env,
+            &ScheduleOptions {
+                genesis_size: 10,
+                peers_to_remove: 3,
+                // high probability - peers will be removed in close succession
+                prob_remove: 0.2,
+                ..Default::default()
+            },
+        );
```
Comment (madadam) [inline tests/integration_tests.rs]: Maybe use the explicit schedule to make sure those removes really happen at the same time? Perhaps within the same step even?

File: tests/integration_tests.rs [hunk 4]
```diff
@@ -247,6 +250,70 @@ mod test {
...
+                // section.
+                (10, Fail(PeerId::new("Alice"))),
+                (20, RemovePeer(PeerId::new("Bob"))),
+                (200, AddPeer(PeerId::new("Hank"))),
+                (700, RemovePeer(PeerId::new("Carol"))),
+                (850, Opaque(Transaction::new("whatever"))),
+            ],
+        };
+        let schedule =
+            Schedule::from_observation_schedule(&mut env, &Default::default(), obs_schedule);
```
Comment (madadam) [inline tests/integration_tests.rs]: Super-nitpick, but I'd replace `Default::default()` with `ScheduleOptions::default()`, so it's immediately obvious what is it we are passing there. But feel free to ignore if you disagree.

**Pull request merged**